Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs

```diff
@@ -157,64 +157,68 @@ private string GetChannelNumber(string extInf, Dictionary<string, string> attrib
             var nameInExtInf = nameParts.Length > 1 ? nameParts.Last().Trim() : null;
 
             string numberString = null;
+            string attributeValue;
+            double doubleValue;
 
-            // Check for channel number with the format from SatIp
-            // #EXTINF:0,84. VOX Schweiz
-            // #EXTINF:0,84.0 - VOX Schweiz
-            if (!string.IsNullOrWhiteSpace(nameInExtInf))
+            if (attributes.TryGetValue("tvg-chno", out attributeValue))
             {
-                var numberIndex = nameInExtInf.IndexOf(' ');
-                if (numberIndex > 0)
+                if (double.TryParse(attributeValue, NumberStyles.Any, CultureInfo.InvariantCulture, out doubleValue))
```

Comment:
why do you parse it as `double`? I thought channel numbers should be `int`s...

Reply:
previous implementation already parsed them into doubles for validation puposes only. Honestly didnt think about changing this to int s while agreeing that a int should be enough by far. Ill change this.

Reply:
If channel numbers there can have values like `11.2` then no, you shouldn't change :)
Essentially if it isn't used elsewhere let's leave it be for now to be on the safe side.